src/pages/CourseForm.js

```diff
@@ -0,0 +1,219 @@
+import {
+  TextboxWithLabels,
+  TextboxWithoutLabels,
+  DateInputWithLabels,
+} from "../components/Textbox";
+import { FileUpload } from "../components/FileUpload";
+// import { AssignCourseCard } from "../components/Card";
+import { push, ref, set } from "firebase/database";
+import { db } from "../firebase";
+import { useState, useEffect } from "react";
+import { AlertError, AlertSuccess } from "../components/Alerts";
+
+const generateRandomAlphanumeric = (length) => {
+  const characters =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+  let result = "";
+  for (let i = 0; i < length; i++) {
+    const randomIndex = Math.floor(Math.random() * characters.length);
+    result += characters.charAt(randomIndex);
+  }
+  return result;
+};
+
+const generateCourseID = () => {
+  const currentDate = new Date();
+  const year = currentDate.getFullYear();
+  const month = String(currentDate.getMonth() + 1).padStart(2, "0");
+  const day = String(currentDate.getDate()).padStart(2, "0");
+  const randomPart = generateRandomAlphanumeric(10);
+
+  const courseID = `${year}${month}${day}${randomPart}`;
+  return courseID;
+};
+
+const extractGid = (link) => {
+  const parts = link.split("#gid=");
+
+  if (parts.length > 1) {
+    return parts[1]; // Extract the GID after #gid=
+  } else {
+    throw new Error("Invalid Google Sheets link format");
+  }
+};
+
+export const CourseForm = () => {
+  const [quizLink, setQuizLink] = useState("");
+  const [courseTitle, setCourseTitle] = useState("");
+  const [courseDescription, setCourseDescription] = useState("");
+  const [courseID, setCourseID] = useState("");
+  const [gid, setGid] = useState("");
+  const [gidValue, setGidValue] = useState("");
+  const [dueDate, setDueDate] = useState("");
+  const coursesRef = ref(db, `courses`);
+  const [showSuccessAlert, setShowSuccessAlert] = useState(false);
+  const [showErrorAlert, setShowErrorAlert] = useState(false);
+
+  useEffect(() => {
+    getCourseID();
+  }, []);
+
+  const getCourseID = () => {
+    setCourseID(generateCourseID());
+  };
+
+  const writeData = async (e) => {
+    e.preventDefault();
+    try {
+      const newCoursesRef = push(coursesRef);
+      await set(newCoursesRef, {
+        createdDate: new Date().toLocaleString(),
+        courseTitle: courseTitle,
+        courseDescription: courseDescription,
+        quizLink: quizLink,
+        gid: gid,
+        courseID: courseID,
+        dueDate: dueDate,
+      });
+      window.scrollTo(0, 0); //scroll to the top after submission
+      setShowSuccessAlert(true);
```

Comment:
This would also happen if it doesn't succeed. Maybe wrap the `set` in a try catch block to handle an error scenario?

Reply:
Didn't see the try catch, sorry!